app/controllers/articles_controller.rb

```diff
@@ -37,7 +38,12 @@ def index
   def create
     @current_user = current_user
     @article = @current_user.articles.new(article_params)
-
+    
+    if params[:article][:location]
+      api_request = HTTP.get(location_api_url).as_json
+      @article.location = get_only_location(JSON.parse(api_request["body"][0]))
```

Comment:
Having the location API methods live in a view helper is perhaps not the best place. Code in `/app/helpers` should really just be focused on making your life easier in the view layer.

I'd recommend moving all of this functionality into a new class within the `/app/lib` directory. Would be cool if the class took in an IP address on initialize and had a single method to spit out the location data. So your code here could look like: `@article.location = Geolocator.new(request.remote_ip).location` (name the things whatever you think is best), and the class itself would handle all of the nitty gritty around API interactions and slicing out the desired data.

Benefit of something like that would be that all the location code would be encapsulated in one place, you could minimize the impact this has on your controllers, and you could unit test the geolocating class if you felt inclined.